src/storage/src/render/mod.rs

```diff
@@ -290,35 +289,46 @@ pub fn build_ingestion_dataflow<A: Allocate>(
                     .enable_multi_worker_storage_persist_sink
                 {
                     tracing::info!(
-                        "timely-{worker_id} rendering {target} with multi-worker persist_sink",
+                        "timely-{worker_id} rendering {export_id} with multi-worker persist_sink",
                     );
                     crate::render::multi_worker_persist_sink::render(
                         into_time_scope,
-                        target,
-                        export.storage_metadata,
+                        export_id,
+                        export.storage_metadata.clone(),
                         source_data,
                         storage_state,
                         metrics,
                         export.output_index,
                     )
                 } else {
                     tracing::info!(
-                        "timely-{worker_id} rendering {target} with single-worker persist_sink",
+                        "timely-{worker_id} rendering {export_id} with single-worker persist_sink",
                     );
                     crate::render::persist_sink::render(
                         into_time_scope,
-                        target,
+                        export_id,
                         export.output_index,
-                        export.storage_metadata,
+                        export.storage_metadata.clone(),
                         source_data,
                         storage_state,
                         metrics,
                     )
                 };
                 tokens.push(token);
+
+                let health_token = crate::source::health_operator(
```

Comment:
You can delay rendering the health operator until after this for loop. Here you can just collect all the health streams in a vector:

```rust
health_streams.push(health);
```

And then render a single health operator with the combined stream:

```rust
use timely::dataflow::operators::Concatenate;

if let Some(health) = health_streams.pop() {
    let combined_health = health.concatenate(health_streams);
    // A single operator for all the data
    let health_token = crate::source::health_operator(..., combined_health);
}
```

This is desirable since each health operator competes with every other one to compare and append to the shard so it's beneficial to have as few of them as possible. In fact in a future refactor we should make it a single one per `clusterd` but for now we should at least keep their number the same.

Reply:
ty for this pointer; this actually means we don't need to partition out the stream at all and can just pass it through and then "demux" it in the operator itself by just keeping the `output_index` intact. Makes the operator more complex because it maintains state for each output but I don't think it's _too_ bad. Am going to merge this if it's green and can refactor later if it's too ugly to bear.